gremlin-javascript/src/main/javascript/gremlin-javascript/lib/structure/io/binary/internals/BulkSetSerializer.js

```diff
@@ -0,0 +1,121 @@
+      if (buffer === undefined || buffer === null || !(buffer instanceof Buffer)) {
+        throw new Error('buffer is missing');
+      }
+      if (buffer.length < 1) {
+        throw new Error('buffer is empty');
+      }
+
+      if (fullyQualifiedFormat) {
+        const type_code = cursor.readUInt8();
+        len++;
+        cursor = cursor.slice(1);
+        if (type_code !== this.ioc.DataType.BULKSET) {
+          throw new Error('unexpected {type_code}');
+        }
+
+        if (cursor.length < 1) {
+          throw new Error('{value_flag} is missing');
+        }
+        const value_flag = cursor.readUInt8();
+        len++;
+        cursor = cursor.slice(1);
+        if (value_flag === 1) {
+          return { v: null, len };
+        }
+        if (value_flag !== 0) {
+          throw new Error('unexpected {value_flag}');
+        }
+      }
+
+      // {length}
+      let length, length_len;
+      try {
+        ({ v: length, len: length_len } = this.ioc.intSerializer.deserialize(cursor, false));
+        len += length_len;
+        cursor = cursor.slice(length_len);
+      } catch (e) {
+        throw new Error(`{length}: ${e.message}`);
+      }
+      if (length < 0) {
+        throw new Error('{length} is less than zero');
+      }
+
+      // Official GraphBinary 1.0 spec:
+      // If the implementing language does not have a BulkSet object to deserialize into,
+      // this format can be coerced to a List and still be considered compliant with Gremlin.
+      // Simply "expand the bulk" by adding the item to the List the number of times specified by the bulk.
+
+      // {item_0}...{item_n}
+      let v = new Array();
+      for (let i = 0; i < length; i++) {
+        let value, value_len;
+        try {
+          ({ v: value, len: value_len } = this.ioc.anySerializer.deserialize(cursor));
+          len += value_len;
+          cursor = cursor.slice(value_len);
+        } catch (e) {
+          throw new Error(`{item_${i}} value: ${e.message}`);
+        }
+
+        let bulk, bulk_len;
+        try {
+          ({ v: bulk, len: bulk_len } = this.ioc.longSerializer.deserialize(cursor, false));
+          len += bulk_len;
+          cursor = cursor.slice(bulk_len);
+        } catch (e) {
+          throw new Error(`{item_${i}} bulk: ${e.message}`);
+        }
+        if (bulk < 0) {
+          throw new Error(`{item_${i}}: bulk is less than zero`);
+        }
+        if (bulk > 4294967295) {
+          // arrayLength of Array() constructor is expected to be an integer between 0 and 2^32 - 1
+          throw new Error(`{item_${i}}: bulk is greater than 2^32-1`);
+        }
+
+        bulk = Number(bulk); // coersion to Number is required for BigInt if LongSerializerNg is used
+        const item = new Array(bulk).fill(value);
+        v = v.concat(item);
+      }
+
+      return { v, len };
+    } catch (e) {
+      throw this.ioc.utils.des_error({ serializer: this, args: arguments, cursor, msg: e.message });
```

Comment:
As an enhancement, I think we can still make use of the original stack trace by keeping the original error.

Maybe we can create a ticket for it?

Reply:
Please, check [this change](https://github.com/apache/tinkerpop/pull/1547/commits/1b71d7ff8d2f685842e4fcec56ce643ebb41481f).